gtest/src/manager.rs

```diff
@@ -30,17 +30,82 @@ use gear_core::{
 use std::collections::{BTreeMap, VecDeque};
 use std::time::{SystemTime, UNIX_EPOCH};
 
-#[derive(Clone, Debug)]
-pub(crate) enum ProgramState {
-    Initialized,
-    Uninitialized(Option<MessageId>),
-    FailedInitialization,
+#[derive(Debug)]
+pub(crate) enum Actor {
+    Initialized(Program),
+    // Contract: program is always `Some`, option is used to take ownership
+    Uninitialized(Option<MessageId>, Option<Program>),
+    Dormant,
+}
+
+impl Actor {
+    pub(crate) fn new(prog: Program) -> Self {
+        Actor::Uninitialized(None, Some(prog))
+    }
+
+    // #Panics
+    // If actor is initialized or dormant
+    fn transform_to_initialized(&mut self) {
```

Comment:
I guess `set_` is ok here too.

```suggestion
    fn set_to_initialized(&mut self) {
```

Reply:
or more simpler: `fn set_initialized(&mut self)`

Reply:
done